app/scripts/migrations/077-supplements/077-supplements.md

```diff
@@ -0,0 +1,98 @@
+
+The fix in the migration 84 supplement is to delete the `NetworkController.network` property if the `NetworkId` property already exists.
+
+**Migration 86**
+
+Renamed the `NetworkController.provider` property to `providerConfig`. If this migration had not run, the NetworkController would have a `provider` property and
+a `providerConfig` property. If migration 86 later ran on this state, the old (and forgotten) `provider` property could cause the `providerConfig` property to be overwritten,
+affecting the state the user's experience was currently depending on.
+
+The fix in the migration 86 supplement is to delete the `NetworkController.provider` property if the `providerConfig` property already exists.
+
+**Migration 88**
+
+Attempted to change the keys of multiple parts of state related to tokens. In particular, `NftController.allNftContracts`, `NftController.allNfts`, `TokenListController.tokensChainsCache`, `TokensController.allTokens`, `TokensController.allIgnoredTokens` and `TokensController.allDetectedTokens`. All of these objects were keyed by chainId in decimal number form. The migration's
+purpose was to change those decimal chain ID keys to hexadecimal. If migration 77 failed, and then the user added or modified tokens, they could have duplicates within these parts of state:
+some with decimal keys and others with an equivalent hexadecimal key. If the data pointed to by those keys was modified at all, and the migration 88 was later run, the most recent data (under
+the hexadecimal key) could be overwritten by the old data under the decimal key.
+
+The migration 88 supplement fixes this by deleting the properties with decimal keys if an equivalent hexadecimal key exists.
+
+## Migrations that were not supplemented
+
+**Migration 78** was not supplemented because it only deletes data; it does not overwrite data. It's failure to run will have left rogue data in state, but that will be removed when it is run after the migration
+77 fix.
+
+**Migration 79** was not supplemented because it only deletes data; it does not overwrite data.
+
+**Migration 80** was not supplemented because it only deletes data; it does not overwrite data.
+
+**Migration 81** was not supplemented because it modifies data that could only be in state on a flask build. The bug that caused the undefined data in tokenlistcontroller state was present on v14.0.2 and v14.1.0 of
+the controllers, but fixed in v14.2.0 of the controllers. By the time flask was released to prod, controllers was at v25.0
+
+**Migration 83** just builds on migration 82. No additional fix is needed for 83 given that we have the 82 supplement.
+
+**Migration 85** was not supplemented because it only deletes data; it does not overwrite data.
+
+**Migration 87** was not supplemented because it only deletes data; it does not overwrite data.
+
+**Migration 89** just builds on migration 82 and 84. No additional fix is needed for 89 given that we have the 82 and 84 supplement.

```

Comment:
Nit: Maybe worth mentioning why migration 90 needs no supplement as well

Reply:
done in 8787024bea